queue_job/tests/test_delayable.py

```diff
@@ -8,9 +8,20 @@
 
 
 class TestDelayable(unittest.TestCase):
-    def setUp(self):
-        super().setUp()
-        self.recordset = mock.MagicMock(name="recordset")
+    @classmethod
+    def setUpClass(cls):
```

Comment:
IMO this change is useless. This class does not inherit from an odoo class and does not trigger any odoo functionality

Reply:
Well, it avoids redeclaring the case for every test. Indeed not much gain here anyway and the context thing would be useless here, that's true

Reply:
It's a mocked object... changes nothing in terms of perf.
Anyway, I'd keep the setUpClass but I'd drop the ctx stuff because it gives the false idea that is needed for whatever  reason.

Reply:
I agree :+1: Please, @josep-tecnativa :)

Reply:
Well, maybe in terms of "consistency" to not wondering if put it or not.

Reply:
The thing here is that this class test inherits directly from `unittest` so it's quite useless to set that

Reply:
Yes, useless, but consistent to always read the same code everywhere in OCA (I wonder also why this is not using Odoo tests), and with minimum overhead. And the day this is switched to Odoo tests, you already have it.

Reply:
I disagree because you should always keep the setup minimal.
However.... Now that I think of this twice... I fear that since the refactoring of the test suite was done (14, 15, 16) these tests are not auto-discovered anymore (just search for this test class in the build).
Hence, we should use `BaseCase` at list.

Reply:
@josep-tecnativa switch to `BaseCase` or `TransactionCase`, please.